### DISCUSSION TITLE
Soft Intro to APIs

### DISCUSSION THREAD

**Discussion by drteresavasquez** [nss-evening-cohort-14/evening-client-side#204] [Class Notes and Discussion]
After finishing up MVP for Nutshell, we want you to take a slow start into the world of APIs even though you have already been using Firebase as an API, so the only difference here is that you need to change your url for the `.get()`.

Here is a link to the repo we will be using: https://github.com/drteresavasquez/get-started-with-apis/blob/main/README.md
**Start with the first one and work your way down the list as the APIs get harder as you move through them.** 

This is meant to be fun and low stress. So, move at a steady pace, but don't feel like you need to rush.

## Goals
- To become more comfortable and familiar with using Postman for sample calls to APIs
- To become acquainted with different types of API calls and requirements
- To understand that API documentation is the key to successful use of APIs
- To create at least 2 apps using APIs
- HAVE FUN!!

If you have any questions about this project, please ask them here.

**Comment by jrobinson0529** [comment]
"Create cards for all the characters"

**Question**: Are we going to create cards for _every_ character? That would be apparently 933 which would crash users.

**Answer**: Use [_Pagination_](https://the-one-api.dev/documentation#5). This is a good opportunity to put some bootstrap pagination buttons on the page and use some JS to get to the next page!

**Reply by drteresavasquez** [reply]
This is for the LOTR API

**Comment by jrobinson0529** [comment]
### QUICK PSA ###
I was getting 404 error on my API call inside my code even though it ran fine in postman.

If you are at the weather API notice that the API call provided in the documentation for current weather data will copy as ```api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key}```

This however will not work when you make your GET call inside your code, you will need to add ```https://``` as a prefix!

So the call will be ```axios.get(https://api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key})```

**Comment by chiestroud** [comment]
**Flipping cards in Lord of the Rings API**
One of the TODOs for Lord of the Rings API is to create cards for all the characters and flip it to show quotes from character when you click the card.

I was wondering how to flip cards easily and found a really nice tool called 'ReactCardFlip".
You can npm install react-card-flip - documentation is here (https://www.npmjs.com/package/react-card-flip).
Documentation shows examples in Class components, but you can watch this video for 5 minutes to convert them into Hooks and now my cards are FLIPPING!
https://www.youtube.com/watch?v=qrZFo1jLG-4
Just wanted to share this for anyone interested :)

**Reply by drteresavasquez** [reply]
Excellent find!!
